server/src/main/java/io/druid/segment/realtime/RealtimeManager.java

```diff
@@ -294,60 +265,69 @@ public FireDepartmentMetrics getMetrics()
     @Override
     public void run()
     {
-      plumber = initPlumber();
+      initPlumber();
 
       try {
-        Object metadata = plumber.startJob();
-
-        if (fireDepartment.checkFirehoseV2()) {
-          firehoseV2 = initFirehoseV2(metadata);
-          runFirehoseV2(firehoseV2);
-        } else {
-          firehose = initFirehose();
-          runFirehose(firehose);
-        }
+        final Closer closer = Closer.create();
 
+        try {
```

Comment:
```Closer.rethrow()``` throws an IOException, so outer IOException-catching block needs to be copied into wherever ```closer.rethrow()``` and ```closer.close()``` are.